### ISSUE TITLE
unused variable false positive for array index in printf -v

### ISSUE THREAD

**Issue by plmnikulin** [koalaman/shellcheck#1874]
#### For bugs
- Rule Id: SC2034
- My shellcheck version: online
- [X] The rule's wiki page does not already cover this (e.g. https://github.com/koalaman/shellcheck/wiki/SC2034)
- [ ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

Tried in shellcheck.net but did not try to build the latest commit.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
fun() {
        local i=0
        printf -v 'c[i++]' '%s' 'abc'
        printf -v 'c[i++]' '%s' 'def'
        echo "${c[*]}"
}

fun
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 3:
        local i=0
              ^-- SC2034: i appears unused. Verify use (or export if used externally).

$ 
```

#### Here's what I wanted or expected to see:

No warning that `i` is unused.
